pkg/engine/internal/imageverifier.go

```diff
@@ -59,11 +65,29 @@ func HasImageVerifiedAnnotationChanged(ctx engineapi.PolicyContext, log logr.Log
 	}
 	newValue := newResource.GetAnnotations()[engineapi.ImageVerifyAnnotationKey]
 	oldValue := oldResource.GetAnnotations()[engineapi.ImageVerifyAnnotationKey]
-	result := newValue != oldValue
-	if result {
-		log.V(2).Info("annotation mismatch", "oldValue", oldValue, "newValue", newValue, "key", engineapi.ImageVerifyAnnotationKey)
+
+	request := admissionv1.AdmissionRequest{}
+	request.Operation = "UPDATE"
+	request.Object.Raw = []byte(newValue)
+	request.OldObject.Raw = []byte(oldValue)
+
+	resource_ctx := ct.NewContext(jp)
+	resource_ctx.AddRequest(request)
+
+	var objMap map[string]interface{}
+
+	err := json.Unmarshal([]byte(newValue), &objMap)
+	if err != nil {
+		fmt.Println("Error:", err)
 	}
-	return result
+
+	for key, _ := range objMap {
+		isChanged, _ := resource_ctx.HasChanged(key)
+		if isChanged {
+			return true
+		}
+	}
```

Comment:
Not sure this logic will work, as it allows adding an image in the new Object annotation.

It may be best to load both the old and new annotation objects in memory, and then check for each entry in the `old` that the same value exists in the `new`.

We don't need to use the`enginecontext.HasChanged` method as it is an expensive operation.

Reply:
> Not sure this logic will work, as it allows adding an image in the new Object annotation.
> 
> It may be best to load both the old and new annotation objects in memory, and then check for each entry in the `old` that the same value exists in the `new`.
> 
> We don't need to use the`enginecontext.HasChanged` method as it is an expensive operation.

I have tested this approach for a few combinations it was working fine. But I am still not confident about this. 
So I think loading the annotation objects in local memory and then comparing them is a better approach.

Reply:
>It may be best to load both the old and new annotation objects in memory, and then check for each entry in the old that the same value exists in the new.

Not sure I follow. 

This is the log dumped during a rollback, the **new** and **old** values are:

`"newValue"="{\"ghcr.io/hackeramitkumar/test5:app\":true}"`
`"oldValue"="{\"ghcr.io/hackeramitkumar/test5:app2\":true}"`

As you can see there's no intersection between the new and old.

```
I0712 10:02:09.032928       1 imageverifier.go:64] engine.verify "msg"="annotation mismatch" "key"="kyverno.io/verify-images" "new.kind"="Deployment" "new.name"="nginx-deployment" "new.namespace"="default" "newValue"="{\"ghcr.io/hackeramitkumar/test5:app\":true}" "old.kind"="Deployment" "old.name"="nginx-deployment" "old.namespace"="default" "oldValue"="{\"ghcr.io/hackeramitkumar/test5:app2\":true}" "policy.apply"="All" "policy.name"="verify-image" "policy.namespace"="" "rule.name"="verify-image"
```

Am I missing anything?